File: Common/Entity/src/main/java/it/ey/dto/AvvisoDTO.java
package it.ey.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;

@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
public class AvvisoDTO extends BaseDTO {
    private Long id;
    private String tipologiaContenuto;
    private LocalDate dataPubblicazione;
    private String oggetto;
    private String tipologiaAmministrazione;
    private String amministrazione;
    private String messaggio;
    private String stato;
}

File: dfp-piao-fe/src/app/shared/models/classes/avviso-dto.ts
import { TypeErrorEnum } from '../enums/type-error.enum';
import { CampiTecniciDTO } from './campi-tecnici-dto';

export class AvvisoDTO extends CampiTecniciDTO {
  active?: boolean;
  deactivationTime?: string;
  campiModificati?: string;
  testoSezione?: string;
  idPiao?: number;
  statoSezione?: string;
  messageError?: string;
  errorCode?: string;
  typeEnum?: TypeErrorEnum;
  id?: number;
  tipologiaContenuto?: string;
  dataPubblicazione?: string;
  oggetto?: string;
  tipologiaAmministrazione?: string;
  amministrazione?: string;
  messaggio?: string;
  stato?: string;

  // Campi legacy mantenuti per compatibilita durante la transizione API/UI.
  visualizzato?: boolean;
  tipoAvviso?: string;
  titolo?: string;
  intro?: string;
  descrizione?: string;
}

export type CreateAvvisoDTO = AvvisoDTO;
export type UpdateAvvisoDTO = AvvisoDTO;